messaging/handler.go

```diff
@@ -185,8 +185,6 @@ func (h *Handler) servePing(w http.ResponseWriter, r *http.Request) {
 		log.Printf("unable to write client config: %s", err)
 		return
 	}
-
-	w.WriteHeader(http.StatusOK)
```

Comment:
The `json.Encoder` writes to the `ResponseWriter` so the header gets flushed. If we call `WriteHeader()` then we get error messages saying that we're trying to write the header twice.